core/api/service/api_service.cpp

```diff
@@ -60,91 +60,110 @@ namespace kagome::api {
+                                                   response.error().message());
+                          });
+                    }
+                  });
 
-        if (SessionType::kWs == session->type()) {
-          auto session_context =
-              self->storeSessionWithId(session->id(), session);
-          session_context.storage_subscription->setCallback(
-              [wp](uint32_t set_id, SessionPtr &session,
-                   const auto &key,
-                   const auto &data,
-                   const auto &block) {
-                if (auto self = wp.lock()) {
-                  jsonrpc::Value::Array out_data;
-                  out_data.emplace_back(api::makeValue(key));
-                  out_data.emplace_back(api::makeValue(data));
-
-                  /// TODO(iceseer): PRE-475 make event notofication depending
-                  /// in blocks, to butch them in a single message
-
-                  jsonrpc::Value::Struct result;
-                  result["changes"] = std::move(out_data);
-                  result["block"] = api::makeValue(block);
-
-                  jsonrpc::Value::Struct params;
-                  params["result"] = std::move(result);
-                  params["subscription"] = api::makeValue(set_id);
-
-                  self->server_->processJsonData(
-                      params, [&](const std::string &response) {
-                        session->respond(response);
-                      });
-                }
-              });
-
-          session_context.events_subscription->setCallback(
-              [wp](uint32_t set_id, SessionPtr &session, const auto &key, const auto &header) {
-                if (auto self = wp.lock()) {
-                  jsonrpc::Value::Struct params;
-                  params["result"] = api::makeValue(header.get());
-                  params["subscription"] = api::makeValue(set_id);
-
-                  self->server_->processJsonData(
-                      params, [&](const std::string &response) {
+              session_context.events_subscription->setCallback(
+                  [wp](uint32_t set_id,
+                       SessionPtr &session,
+                       const auto &key,
+                       const auto &header) {
+                    if (auto self = wp.lock()) {
+                      jsonrpc::Value::Struct p;
+                      p["result"] = api::makeValue(header.get());
+                      p["subscription"] = api::makeValue(set_id);
+
+                      jsonrpc::Request::Parameters params;
+                      params.push_back(std::move(p));
+                      self->server_->processJsonData(
+                          "chain_newHead", params, [&](const auto &response) {
+                            if (response.has_value())
+                              session->respond(response.value());
+                            else
+                              self->logger_->error("ERROR => {}",
```

Comment:
I guess more details in log message would be better to analising of error

Reply:
PR was not requested for review yet.

Reply:
fixed